// angular2-air-datepicker/esm5/lib/components/monthpicker/monthpicker.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, EventEmitter, Input, Output } from '@angular/core';
import { AirCalendar, AirLanguage } from '../../classes';
var MonthpickerComponent = /** @class */ (function () {
    function MonthpickerComponent() {
        this.setMonth = new EventEmitter();
        this.setYear = new EventEmitter();
        this.yearSelection = new EventEmitter();
    }
    MonthpickerComponent.decorators = [
        { type: Component, args: [{
                    selector: '[monthpicker]',
                    template: "\n    <nav class=\"datepicker--nav\">\n      <div class=\"datepicker--nav-action\" (click)=\"setYear.emit(airCalendar.year - 1)\"><svg><path d=\"M 17,12 l -5,5 l 5,5\"></path></svg></div>\n\n      <div (click)=\"yearSelection.emit()\" class=\"datepicker--nav-title\">{{airCalendar.year}}</div>\n\n      <div class=\"datepicker--nav-action\" (click)=\"setYear.emit(airCalendar.year + 1)\"><svg><path d=\"M 14,12 l 5,5 l -5,5\"></path></svg></div>\n    </nav>\n\n    <div class=\"datepicker--content\">\n      <div class=\"datepicker--months datepicker--body active\">\n        <div class=\"datepicker--cells datepicker--cells-months\">\n          <div *ngFor=\"let month of airLanguage.months; let i=index\"\n               (click)=\"setMonth.emit(i)\"\n               class=\"datepicker--cell datepicker--cell-month\"\n               [ngClass]=\"{ '-current-': airCalendar.year == airCalendar.currentYear && i == airCalendar.currentMonth }\">{{month}}</div>\n        </div>\n      </div>\n    </div>\n  "
                }] }
    ];
    MonthpickerComponent.propDecorators = {
        airCalendar: [{ type: Input }],
        airLanguage: [{ type: Input }],
        setMonth: [{ type: Output }],
        setYear: [{ type: Output }],
        yearSelection: [{ type: Output }]
    };
    return MonthpickerComponent;
}());
export { MonthpickerComponent };
if (false) {
    /** @type {?} */
    MonthpickerComponent.prototype.airCalendar;
    /** @type {?} */
    MonthpickerComponent.prototype.airLanguage;
    /** @type {?} */
    MonthpickerComponent.prototype.setMonth;
    /** @type {?} */
    MonthpickerComponent.prototype.setYear;
    /** @type {?} */
    MonthpickerComponent.prototype.yearSelection;
}
