// angular2-air-datepicker/esm5/lib/components/yearpicker/yearpicker.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, EventEmitter, Input, Output } from '@angular/core';
import { AirCalendar } from '../../classes';
var YearpickerComponent = /** @class */ (function () {
    function YearpickerComponent() {
        this.setYear = new EventEmitter();
        this.years = [];
    }
    /**
     * @return {?}
     */
    YearpickerComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        /** @type {?} */
        var firstYear = this.airCalendar.year - 6;
        this.years = Array.from({ length: 12 }, function (v, k) { return firstYear + k; });
    };
    /**
     * @return {?}
     */
    YearpickerComponent.prototype.next = /**
     * @return {?}
     */
    function () {
        for (var i = 0; i < this.years.length; i++) {
            this.years[i] += 10;
        }
    };
    /**
     * @return {?}
     */
    YearpickerComponent.prototype.previous = /**
     * @return {?}
     */
    function () {
        for (var i = 0; i < this.years.length; i++) {
            this.years[i] -= 10;
        }
    };
    YearpickerComponent.decorators = [
        { type: Component, args: [{
                    selector: '[yearpicker]',
                    template: "\n    <nav class=\"datepicker--nav\">\n      <div class=\"datepicker--nav-action\" (click)=\"previous()\"><svg><path d=\"M 17,12 l -5,5 l 5,5\"></path></svg></div>\n\n      <div class=\"datepicker--nav-title\">{{years[1]}} - {{years[years.length - 2]}}</div>\n\n      <div class=\"datepicker--nav-action\" (click)=\"next()\"><svg><path d=\"M 14,12 l 5,5 l -5,5\"></path></svg></div>\n    </nav>\n\n    <div class=\"datepicker--content\">\n      <div class=\"datepicker--cells datepicker--cells-years\">\n        <div *ngFor=\"let year of years; let i=index\" (click)=\"setYear.emit(year)\" class=\"datepicker--cell datepicker--cell-year\"\n             [ngClass]=\"{ '-current-': year == airCalendar.currentYear, '-other-decade-': i == 0 || i == years.length - 1 }\">{{year}}</div>\n      </div>\n    </div>\n  "
                }] }
    ];
    YearpickerComponent.propDecorators = {
        airCalendar: [{ type: Input }],
        setYear: [{ type: Output }]
    };
    return YearpickerComponent;
}());
export { YearpickerComponent };
if (false) {
    /** @type {?} */
    YearpickerComponent.prototype.airCalendar;
    /** @type {?} */
    YearpickerComponent.prototype.setYear;
    /** @type {?} */
    YearpickerComponent.prototype.years;
}
