// angular2-air-datepicker/esm2015/lib/components/yearpicker/yearpicker.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, EventEmitter, Input, Output } from '@angular/core';
import { AirCalendar } from '../../classes';
export class YearpickerComponent {
    constructor() {
        this.setYear = new EventEmitter();
        this.years = [];
    }
    /**
     * @return {?}
     */
    ngOnInit() {
        /** @type {?} */
        const firstYear = this.airCalendar.year - 6;
        this.years = Array.from({ length: 12 }, (v, k) => firstYear + k);
    }
    /**
     * @return {?}
     */
    next() {
        for (let i = 0; i < this.years.length; i++) {
            this.years[i] += 10;
        }
    }
    /**
     * @return {?}
     */
    previous() {
        for (let i = 0; i < this.years.length; i++) {
            this.years[i] -= 10;
        }
    }
}
YearpickerComponent.decorators = [
    { type: Component, args: [{
                selector: '[yearpicker]',
                template: `
    <nav class="datepicker--nav">
      <div class="datepicker--nav-action" (click)="previous()"><svg><path d="M 17,12 l -5,5 l 5,5"></path></svg></div>

      <div class="datepicker--nav-title">{{years[1]}} - {{years[years.length - 2]}}</div>

      <div class="datepicker--nav-action" (click)="next()"><svg><path d="M 14,12 l 5,5 l -5,5"></path></svg></div>
    </nav>

    <div class="datepicker--content">
      <div class="datepicker--cells datepicker--cells-years">
        <div *ngFor="let year of years; let i=index" (click)="setYear.emit(year)" class="datepicker--cell datepicker--cell-year"
             [ngClass]="{ '-current-': year == airCalendar.currentYear, '-other-decade-': i == 0 || i == years.length - 1 }">{{year}}</div>
      </div>
    </div>
  `
            }] }
];
YearpickerComponent.propDecorators = {
    airCalendar: [{ type: Input }],
    setYear: [{ type: Output }]
};
if (false) {
    /** @type {?} */
    YearpickerComponent.prototype.airCalendar;
    /** @type {?} */
    YearpickerComponent.prototype.setYear;
    /** @type {?} */
    YearpickerComponent.prototype.years;
}
