// angular2-air-datepicker/esm5/lib/classes/options.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
var AirOptions = /** @class */ (function () {
    function AirOptions(options) {
        if (options === void 0) { options = (/** @type {?} */ ({})); }
        this.timepicker = !!options.timepicker;
        this.format12h = !!options.format12h;
        this.fullDays = !!options.fullDays;
        this.language = options.language || 'en';
        this.hourStep = options.hourStep || 1;
        this.minuteStep = options.minuteStep || 1;
        this.enabledDateRanges = options.enabledDateRanges || [];
    }
    /**
     * @param {?} date
     * @return {?}
     */
    AirOptions.prototype.isDisabled = /**
     * @param {?} date
     * @return {?}
     */
    function (date) {
        var e_1, _a;
        try {
            for (var _b = tslib_1.__values(this.enabledDateRanges), _c = _b.next(); !_c.done; _c = _b.next()) {
                var dateRange = _c.value;
                if (date >= dateRange.start && date <= dateRange.end) {
                    return false;
                }
            }
        }
        catch (e_1_1) { e_1 = { error: e_1_1 }; }
        finally {
            try {
                if (_c && !_c.done && (_a = _b.return)) _a.call(_b);
            }
            finally { if (e_1) throw e_1.error; }
        }
        return !!this.enabledDateRanges.length;
    };
    return AirOptions;
}());
export { AirOptions };
if (false) {
    /** @type {?} */
    AirOptions.prototype.timepicker;
    /** @type {?} */
    AirOptions.prototype.format12h;
    /** @type {?} */
    AirOptions.prototype.fullDays;
    /** @type {?} */
    AirOptions.prototype.language;
    /** @type {?} */
    AirOptions.prototype.hourStep;
    /** @type {?} */
    AirOptions.prototype.minuteStep;
    /** @type {?} */
    AirOptions.prototype.enabledDateRanges;
}
/**
 * @record
 */
export function DateRange() { }
if (false) {
    /** @type {?} */
    DateRange.prototype.start;
    /** @type {?} */
    DateRange.prototype.end;
}
