// ng-maphilight/esm5/lib/maphilight.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input, ElementRef, KeyValueDiffers, } from '@angular/core';
import { Subject } from 'rxjs';
var MaphilightComponent = /** @class */ (function () {
    function MaphilightComponent(elRef, differs) {
        this.elRef = elRef;
        this.events = {
            imgLoaded: new Subject(),
            updateMaphilight: new Subject(),
        };
        this.el = this.elRef.nativeElement;
        this.configDiffer = differs.find({}).create();
    }
    /**
     * @return {?}
     */
    MaphilightComponent.prototype.ngAfterViewInit = /**
     * @return {?}
     */
    function () {
        var _this = this;
        this.img = this.el.querySelector('img');
        this.map = this.el.querySelector('map');
        /** @type {?} */
        var id = this.el.getAttribute('id');
        this.img.setAttribute('usemap', '#' + id + ".map");
        this.map.setAttribute('id', id + ".map");
        this.map.setAttribute('name', id + ".map");
        // Wait until image is loaded. Otherwise the $(img).height() in maphilight.js may return 0,
        // which causes mouseover effect to break.
        // See also  https://stackoverflow.com/questions/16084374/jquery-width-and-height-return-0-for-img-element
        this.img.onload = (/**
         * @return {?}
         */
        function () {
            _this.events.imgLoaded.next();
            _this.updateMaphilight();
        });
    };
    /**
     * @return {?}
     */
    MaphilightComponent.prototype.updateMaphilight = /**
     * @return {?}
     */
    function () {
        if (!this.img)
            return;
        this.events.updateMaphilight.next();
        window.jQuery(this.img).maphilight(this.configToApply());
    };
    ;
    /**
     * @return {?}
     */
    MaphilightComponent.prototype.ngDoCheck = /**
     * @return {?}
     */
    function () {
        /** @type {?} */
        var changes = this.configDiffer.diff(this.config);
        if (changes) {
            this.updateMaphilight();
        }
    };
    /**
     * @private
     * @return {?}
     */
    MaphilightComponent.prototype.configToApply = /**
     * @private
     * @return {?}
     */
    function () {
        /** @type {?} */
        var config = Object.assign({}, this.config);
        if (config.strokeColor && config.strokeColor[0] == '#')
            config.strokeColor = config.strokeColor.substr(1);
        if (config.fillColor && config.fillColor[0] == '#')
            config.fillColor = config.fillColor.substr(1);
        if (config.shadowColor && config.shadowColor[0] == '#')
            config.shadowColor = config.shadowColor.substr(1);
        return config;
    };
    MaphilightComponent.decorators = [
        { type: Component, args: [{
                    selector: 'maphilight',
                    template: "<ng-content>\n</ng-content>\n",
                    styles: [""]
                }] }
    ];
    /** @nocollapse */
    MaphilightComponent.ctorParameters = function () { return [
        { type: ElementRef },
        { type: KeyValueDiffers }
    ]; };
    MaphilightComponent.propDecorators = {
        config: [{ type: Input }]
    };
    return MaphilightComponent;
}());
export { MaphilightComponent };
if (false) {
    /** @type {?} */
    MaphilightComponent.prototype.config;
    /** @type {?} */
    MaphilightComponent.prototype.configDiffer;
    /** @type {?} */
    MaphilightComponent.prototype.el;
    /** @type {?} */
    MaphilightComponent.prototype.img;
    /** @type {?} */
    MaphilightComponent.prototype.map;
    /** @type {?} */
    MaphilightComponent.prototype.events;
    /**
     * @type {?}
     * @private
     */
    MaphilightComponent.prototype.elRef;
    /* Skipping unhandled member: ;*/
}
